### PR TITLE
PLATUI-156: Functionality and tests for copying code examples to clipboard

### PR DESCRIPTION
**Pull request by silasl**
* Add markup and styles for Copy button
* Added Clipboard.js dependency
* JS to deselect text and change button text
* tests for 👆

### CODE REVIEW

File: application/__tests__/example.macro.integration.js [hunk 1]
```diff
@@ -8,21 +8,47 @@ describe('Example page rendering', () => {
...
+      const buttonText = await page.$eval(buttonSelector, btn => btn.textContent)
+      expect(buttonText).toBe('Copied')
+    })
+
+    it('should reset text to `Copy` after an interval', async () => {
+      let buttonText
+      await new Promise(resolve => setTimeout(async () => {
+        buttonText = await page.$eval(buttonSelector, btn => btn.textContent)
+        resolve()
+      }, 5000))
```
Comment (matcarey) [inline application/__tests__/example.macro.integration.js]: We shouldn't put real delays into tests, this should be mockable :)

File: application/scss/components/_copy-button.scss [hunk 2]
```diff
@@ -0,0 +1,25 @@
+.app-link--copy {
+  &:link,
+  &:active,
+  &:visited,
+  &:hover {
+    @include govuk-font(16);
+    position: absolute;
```
Comment (matcarey) [inline application/scss/components/_copy-button.scss]: This relies on the implicit `position` from `_tabs.scss`, is that safe/reliable enough?

File: application/__tests__/example.macro.test.js [hunk 3]
```diff
@@ -116,7 +123,14 @@ describe('Single page example macro english', () => {
     })
   })
 
-  test('Should include the escaped HTML markup from the examples', () => {
+  it('should have a button to copy Nunjucks code examples', () => {
+    const codeContainer = document.querySelector('#example-default-nunjucks')
+    const copyButton = codeContainer.querySelector('.app-link--copy')
+    expect(copyButton).not.toBeNull()
+    expect(copyButton.text).toBe('Copy')
```
Comment (matcarey) [inline application/__tests__/example.macro.test.js]: Can we test that the (correct) contents is copied too?

File: application/__tests__/example.macro.integration.js (hunk 1 above)
Comment (silasl) [reply application/__tests__/example.macro.integration.js]: AS this is an integration test mocking a delay is overly complex and probably not desirable

File: application/__tests__/example.macro.test.js (hunk 3 above)
Comment (silasl) [reply application/__tests__/example.macro.test.js]: We don't need to test the functionality of the third party component however we are testing it's instantiation by testing the 'onSuccess' method

**Pull request merged**